src/middlewared/middlewared/worker.py

```diff
@@ -94,17 +87,7 @@ def set_progress(self, percent, description=None, extra=None):
 
 def main_worker(*call_args):
     global MIDDLEWARE
-    loop = asyncio.get_event_loop()
-    coro = MIDDLEWARE._run(*call_args)
-    try:
-        res = loop.run_until_complete(coro)
-    except SystemExit:
```

Comment:
I think we should still catch SystemExit? Arbitrary methods could still raise it?